archive: Introducing TOML archive in favour of INI archive

KristianIvarsson opened the pull request:
A new archive and buffer type for the TOML protocol and dispatches for both .toml and .ini files along with TOML and INI buffers

lazan commented on middleware/common/source/serialize/toml.cpp:
```diff
@@ -0,0 +1,388 @@
+   {
+      namespace serialize
+      {
+         namespace toml
```
I think you should collapse these to `common::serialise::toml` 
And do the explicit template instantiation at the end as ` template struct  common::serialize::reader::create::Registration< toml::local::reader::Implementation>:` and so on.

lazan commented on middleware/common/source/serialize/toml.cpp:
```diff
@@ -0,0 +1,388 @@
+                           {
+                              auto node = m_stack.back()->as_table()->find( name);
+
+                              if(node != m_stack.back()->as_table()->end())
```
`if(<ws>...`

lazan commented on middleware/common/include/common/serialize/toml.h:
```diff
@@ -36,9 +36,16 @@ namespace casual
             serialize::Writer writer();
 
-         } // ini
+         } // toml
```
I think you should collapse the ns

lazan commented on middleware/common/include/common/serialize/service/protocol/implementation.h:
```diff
@@ -57,48 +57,48 @@ namespace casual
       };
 
-      struct Ini : public Base
+      struct Xml : public Base
```
Did you rearrange these based on alphabetical order?

lazan reviewed (approved):
Looking good. Hade a few minor suggestions.

KristianIvarsson committed:
archive: Introducing TOML archive in favour of INI archive resolves #594

A new archive and buffer type for the TOML protocol and dispatches for
both .toml and .ini files along with TOML buffers
